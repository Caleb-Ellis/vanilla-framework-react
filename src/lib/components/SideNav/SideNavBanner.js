import React from 'react';
import PropTypes from 'prop-types';

class SideNavBanner extends React.Component {
  render() {
    const {
      href, logo, onClick, open, tagline, title,
    } = this.props;

    const Tag = href ? 'a' : 'div';

    return (
      <header id="sidenav" role="banner">
        <div className="p-navigation__banner">
          <div className="row">
            <div className="mobile-col-2">
              <div className="p-navigation__logo">
                <Tag className="p-navigation__link" href={href}>
                  {(logo.src ?
                    <img className="p-navigation__image" src={logo.src} alt={logo.alt} /> : title
                  )}
                </Tag>
              </div>
              {tagline && <span className="p-navigation__tagline">{tagline}</span>}
            </div>
            <div className="sidebar__cta mobile-col-2 u-align--right">
              <ul className="p-inline-list u-no-margin--top">
                <li className="p-inline-list__item">
                  {/* eslint-disable */}
                  <i
                    className={`p-icon--menu ${open ? 'u-hide' : ''}`}
                    onClick={onClick}
                    aria-hidden="false"
                  />
                  <i
                    className={`p-icon--close ${open ? '' : 'u-hide'}`}
                    onClick={onClick}
                    aria-hidden="true"
                  />
                </li>
              </ul>
            </div>
          </div>
        </div>
      </header>
    );
  }
}

SideNavBanner.defaultProps = {
  href: null,
  logo: { src: null, alt: '' },
  onClick: () => 1,
  open: false,
  tagline: '',
  title: null,
};

SideNavBanner.propTypes = {
  href: PropTypes.string,
  logo: PropTypes.shape({ src: PropTypes.string, alt: PropTypes.string }),
  onClick: PropTypes.func,
  open: PropTypes.bool,
  tagline: PropTypes.string,
  title: PropTypes.string,
};

SideNavBanner.displayName = 'SideNavBanner';

export default SideNavBanner;
