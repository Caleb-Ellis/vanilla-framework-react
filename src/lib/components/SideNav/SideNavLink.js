import React from 'react';
import PropTypes from 'prop-types';
import getClassName from '../../utils/getClassName';

const SideNavLink = (props) => {
  const {
    children, href, label, selected,
  } = props;

  const className = getClassName({
    sidebar__link: true,
    'is-selected': selected,
  });

  return (
    <li>
      <a className={className} href={href}>
        {label}
        {children}
      </a>
    </li>
  );
};

SideNavLink.defaultProps = {
  children: null,
  selected: false,
};

SideNavLink.propTypes = {
  children: PropTypes.node,
  label: PropTypes.string.isRequired,
  href: PropTypes.string.isRequired,
  selected: PropTypes.bool,
};

SideNavLink.displayName = 'SideNavLink';

export default SideNavLink;
