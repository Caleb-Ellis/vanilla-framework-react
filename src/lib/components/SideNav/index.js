export { default as SideNav } from './SideNav';
export { default as SideNavBanner } from './SideNavBanner';
export { default as SideNavGroup } from './SideNavGroup';
export { default as SideNavLink } from './SideNavLink';
