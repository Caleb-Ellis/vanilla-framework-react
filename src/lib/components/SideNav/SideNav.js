import React from 'react';

import SideNavBanner from './SideNavBanner';

class SideNav extends React.Component {
  constructor() {
    super();
    this.handleMenuClick = this.handleMenuClick.bind(this);

    this.state = { menuOpen: false };
  }

  handleMenuClick() {
    const { menuOpen } = this.state;
    this.setState({ menuOpen: !menuOpen });
  }

  render() {
    const { children } = this.props;
    const { menuOpen } = this.state;
    const content = [];
    let banner;

    React.Children.forEach(children, (child) => {
      if (child.type === SideNavBanner) {
        banner = React.cloneElement(child, {
          ...this.props,
          onClick: this.handleMenuClick,
          open: this.state.menuOpen,
        });
      } else {
        content.push(child);
      }
    });

    return (
      <div className="p-navigation--sidebar">
        { banner }
        <div className={`sidebar__content u-no-margin--top ${menuOpen ? '' : 'u-hide'}`}>
          <ul className="p-list">
            { content }
          </ul>
        </div>
      </div>
    );
  }
}

SideNav.defaultProps = {
  children: null,
};

SideNav.propTypes = {
  children: (props, propName, componentName) => {
    const prop = props[propName];
    let error = null;
    let count = 0;

    React.Children.forEach(prop, (child) => {
      if (child.type === SideNavBanner) {
        count += 1;
      }
    });

    if (count !== 1) {
      error = new Error(`${componentName} should have exactly one child of type "SideNavBanner".`);
    }

    return error;
  },
};

SideNav.displayName = 'SideNav';

export default SideNav;
