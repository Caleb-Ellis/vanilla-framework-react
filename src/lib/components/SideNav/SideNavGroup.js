import React from 'react';
import PropTypes from 'prop-types';
import getClassName from '../../utils/getClassName';

import SideNavLink from './SideNavLink';

class SideNavGroup extends React.Component {
  constructor(props) {
    super(props);
    this.onClick = this.onClick.bind(this);

    this.state = { selected: props.selected };
  }

  onClick(e) {
    e.preventDefault();
    const { selected } = this.state;
    this.setState({ selected: !selected });
  }

  render() {
    const { children, href, label } = this.props;
    const { selected } = this.state;

    const className = getClassName({
      sidebar__link: true,
      'is-selected': selected,
    });

    return (
      <li>
        <a className={className} href={href} onClick={this.onClick}>
          { label }
          <i className="p-icon--plus" />
          <i className="p-icon--minus" />
        </a>
        <ul className="sidebar__second-level">
          { children }
        </ul>
      </li>
    );
  }
}

SideNavGroup.defaultProps = {
  children: null,
  href: '#',
  selected: false,
};

SideNavGroup.propTypes = {
  children: (props, propName, componentName) => {
    const prop = props[propName];
    let error = null;

    React.Children.forEach(prop, (child) => {
      if (child.type !== SideNavLink) {
        error = new Error(`${componentName} children should be of type "SideNavLink".`);
      }
    });

    return error;
  },
  href: PropTypes.string,
  label: PropTypes.string.isRequired,
  selected: PropTypes.bool,
};

SideNavGroup.displayName = 'SideNavGroup';

export default SideNavGroup;
