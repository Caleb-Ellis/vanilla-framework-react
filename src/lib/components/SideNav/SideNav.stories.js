import React from 'react';
import { storiesOf } from '@storybook/react';
import { number, text } from '@storybook/addon-knobs';
import { withInfo } from '@storybook/addon-info';

import Link from '../Link/Link';
import List from '../List/List';
import ListItem from '../List/ListItem';
import SideNav from './SideNav';
import SideNavBanner from './SideNavBanner';
import SideNavGroup from './SideNavGroup';
import SideNavLink from './SideNavLink';

const options = {
  range: true,
  min: 3,
  max: 12,
  step: 1,
};

storiesOf('SideNav', module)
  .add('Text Banner',
    withInfo('The SideNav component allows for navigation from a collapsing menu on the side of the page. It requires a SideNavBanner which contains a title/logo, optional tagline, and a space for the small screen burger menu. SideNavLinks can be placed directly inside SideNav, or inside a SideNavGroup to keep them under section titles. This example will expand to fill the space available to it so it needs to be used in conjunction with the grid to set the layout.')(() => (
      <div className={`col-${number('Column size', 12, options)}`}>
        <SideNav>
          <SideNavBanner
            href="#"
            tagline={text('Tagline', 'Lorem ipsum')}
            title={text('Title', 'Vanilla')}
          />
          <SideNavGroup label="Section title 1">
            <SideNavLink href="#" label="Second level 1" />
            <SideNavLink href="#" label="Second level 2" />
            <SideNavGroup label="Second level 3">
              <SideNavLink href="#" label="Third level 1" />
              <SideNavLink href="#" label="Third level 2" />
              <SideNavLink href="#" label="Third level 3" />
              <SideNavLink href="#" label="Third level 4" />
              <SideNavLink href="#" label="Third level 5" />
            </SideNavGroup>
            <SideNavLink href="#" label="Second level 4" />
            <SideNavLink href="#" label="Second level 5" />
          </SideNavGroup>
          <SideNavGroup label="Section title 2">
            <SideNavLink href="#" label="Second level 1" />
            <SideNavLink href="#" label="Second level 2" />
            <SideNavGroup label="Second level 3">
              <SideNavLink href="#" label="Third level 1" />
              <SideNavLink href="#" label="Third level 2" />
              <SideNavLink href="#" label="Third level 3" />
              <SideNavLink href="#" label="Third level 4" />
              <SideNavLink href="#" label="Third level 5" />
            </SideNavGroup>
            <SideNavLink href="#" label="Second level 4" />
            <SideNavLink href="#" label="Second level 5" />
          </SideNavGroup>
          <SideNavGroup label="Section title 3">
            <SideNavLink href="#" label="Second level 1" />
            <SideNavLink href="#" label="Second level 2" />
            <SideNavGroup label="Second level 3">
              <SideNavLink href="#" label="Third level 1" />
              <SideNavLink href="#" label="Third level 2" />
              <SideNavLink href="#" label="Third level 3" />
              <SideNavLink href="#" label="Third level 4" />
              <SideNavLink href="#" label="Third level 5" />
            </SideNavGroup>
            <SideNavLink href="#" label="Second level 4" />
            <SideNavLink href="#" label="Second level 5" />
          </SideNavGroup>
          <SideNavLink href="#" label="First level 1" />
          <hr />
          <List>
            <ListItem>
              <Link external className="sidebar__link" href="#a">Supplementary link 1</Link>
            </ListItem>
            <ListItem>
              <Link external className="sidebar__link" href="#a">Supplementary link 2</Link>
            </ListItem>
          </List>
        </SideNav>
      </div>),
    ),
  )

  .add('Logo Banner',
    withInfo('A logo object prop can also be passed to SideNavBanner, which will replace a simple text banner.')(() => (
      <div className={`col-${number('Column size', 12, options)}`}>
        <SideNav>
          <SideNavBanner
            href="#"
            logo={{
              src: text('Logo src', 'https://assets.ubuntu.com/v1/d96d86b5-vanilla_black-orange_hex.svg'),
              alt: '',
            }}
            tagline={text('Tagline', 'Documentation')}
          />
          <SideNavGroup label="Basics">
            <SideNavLink href="#" label="Color" />
            <SideNavLink href="#" label="Font" />
            <SideNavLink href="#" label="Reset" />
            <SideNavLink href="#" label="Spacing" />
          </SideNavGroup>
          <SideNavGroup label="Global layout">
            <SideNavLink href="#" label="Breakpoints" />
            <SideNavLink href="#" label="Grid" />
            <SideNavLink href="#" label="Strip" />
            <SideNavLink href="#" label="Structure" />
          </SideNavGroup>
          <SideNavGroup label="One-off layout">
            <SideNavLink href="#" label="Alignment" />
            <SideNavLink href="#" label="Clearfix" />
            <SideNavLink href="#" label="Equal height" />
            <SideNavLink href="#" label="Floats" />
            <SideNavLink href="#" label="Hide" />
            <SideNavLink href="#" label="Image position" />
            <SideNavLink href="#" label="Margin collapse" />
            <SideNavLink href="#" label="Off-screen elements" />
            <SideNavLink href="#" label="Padding collapse" />
            <SideNavLink href="#" label="Show" />
            <SideNavLink href="#" label="Vertically center" />
          </SideNavGroup>
          <SideNavGroup label="Navigation">
            <SideNavLink href="#" label="Accordion" />
            <SideNavLink href="#" label="Aside" />
            <SideNavLink href="#" label="Breadcrumbs" />
            <SideNavLink href="#" label="Contextual menu" />
            <SideNavLink href="#" label="Footer" />
            <SideNavLink href="#" label="Global navigation" />
            <SideNavLink href="#" label="Pagination" />
            <SideNavLink href="#" label="Tabs" />
          </SideNavGroup>
          <SideNavGroup label="Page structure">
            <SideNavLink href="#" label="Cards" />
            <SideNavLink href="#" label="Divider list" />
            <SideNavLink href="#" label="Heading icon" />
            <SideNavLink href="#" label="Images" />
            <SideNavLink href="#" label="Inline images" />
            <SideNavLink href="#" label="Lists" />
            <SideNavLink href="#" label="Matrix" />
            <SideNavLink href="#" label="Media object" />
            <SideNavLink href="#" label="Muted heading" />
          </SideNavGroup>
          <SideNavGroup label="Text">
            <SideNavLink href="#" label="Code" />
            <SideNavLink href="#" label="Code numbered" />
            <SideNavLink href="#" label="Code snippet" />
            <SideNavLink href="#" label="Typography" />
            <SideNavLink href="#" label="Quotes" />
          </SideNavGroup>
          <SideNavGroup label="Interactive elements">
            <SideNavLink href="#" label="Buttons" />
            <SideNavLink href="#" label="Forms" />
            <SideNavLink href="#" label="Form layout" />
            <SideNavLink href="#" label="Form validation" />
            <SideNavLink href="#" label="Links" />
            <SideNavLink href="#" label="List tree" />
            <SideNavLink href="#" label="Modal" />
            <SideNavLink href="#" label="Notifications" />
            <SideNavLink href="#" label="Search box" />
            <SideNavLink href="#" label="Slider" />
            <SideNavLink href="#" label="Switch" />
            <SideNavLink href="#" label="Table" />
            <SideNavLink href="#" label="Table sortable" />
            <SideNavLink href="#" label="Table expanding" />
            <SideNavLink href="#" label="Table mobile card" />
            <SideNavLink href="#" label="Tooltips" />
          </SideNavGroup>
          <SideNavGroup label="Media">
            <SideNavLink href="#" label="Animations" />
            <SideNavLink href="#" label="Assets" />
            <SideNavLink href="#" label="Embedded media" />
            <SideNavLink href="#" label="Icons" />
            <SideNavLink href="#" label="Spin" />
          </SideNavGroup>
          <hr />
          <List>
            <ListItem>
              <Link external className="sidebar__link" href="https://vanillaframework.io">vanillaframework.io</Link>
            </ListItem>
          </List>
        </SideNav>
      </div>),
    ),
  );
