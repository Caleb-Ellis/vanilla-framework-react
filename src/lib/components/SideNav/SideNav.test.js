import React from 'react';
import ReactTestRenderer from 'react-test-renderer';

import SideNav from './SideNav';
import SideNavBanner from './SideNavBanner';
import SideNavGroup from './SideNavGroup';
import SideNavLink from './SideNavLink';

describe('<SideNav>', () => {
  it('renders with a text-based SideNavBanner correctly', () => {
    const sidenav = ReactTestRenderer.create(
      <SideNav>
        <SideNavBanner href="#" title="Title" />
      </SideNav>,
    );
    const json = sidenav.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('renders with a logo-based SideNavBanner correctly', () => {
    const sidenav = ReactTestRenderer.create(
      <SideNav>
        <SideNavBanner
          href="#"
          logo={{ src: 'https://assets.ubuntu.com/v1/d96d86b5-vanilla_black-orange_hex.svg', alt: '' }}
        />
      </SideNav>,
    );
    const json = sidenav.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('renders with a single SideNavLink correctly', () => {
    const sidenav = ReactTestRenderer.create(
      <SideNav>
        <SideNavBanner href="#" title="Title" />
        <SideNavLink href="#" label="Link1" />
      </SideNav>,
    );
    const json = sidenav.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('renders with multiple SideNavLinks correctly', () => {
    const sidenav = ReactTestRenderer.create(
      <SideNav>
        <SideNavBanner href="#" title="Title" />
        <SideNavLink href="#" label="Link1" />
        <SideNavLink href="#" label="Link2" />
        <SideNavLink href="#" label="Link3" />
      </SideNav>,
    );
    const json = sidenav.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('renders with a SideNavGroup correctly', () => {
    const sidenav = ReactTestRenderer.create(
      <SideNav>
        <SideNavBanner href="#" title="Title" />
        <SideNavGroup label="Group">
          <SideNavLink href="#" label="Link" />
        </SideNavGroup>
      </SideNav>,
    );
    const json = sidenav.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('renders with multiple SideNavGroups correctly', () => {
    const sidenav = ReactTestRenderer.create(
      <SideNav>
        <SideNavBanner href="#" title="Title" />
        <SideNavGroup label="Group1">
          <SideNavLink href="#" label="Link1" />
        </SideNavGroup>
        <SideNavGroup label="Group2">
          <SideNavLink href="#" label="Link2" />
        </SideNavGroup>
        <SideNavGroup label="Group3">
          <SideNavLink href="#" label="Link3" />
        </SideNavGroup>
      </SideNav>,
    );
    const json = sidenav.toJSON();
    expect(json).toMatchSnapshot();
  });
});
