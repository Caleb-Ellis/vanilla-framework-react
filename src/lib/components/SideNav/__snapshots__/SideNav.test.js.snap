// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SideNav> renders with a SideNavGroup correctly 1`] = `
<div
  className="p-navigation--sidebar"
>
  <header
    id="sidenav"
    role="banner"
  >
    <div
      className="p-navigation__banner"
    >
      <div
        className="row"
      >
        <div
          className="mobile-col-2"
        >
          <div
            className="p-navigation__logo"
          >
            <a
              className="p-navigation__link"
              href="#"
            >
              Title
            </a>
          </div>
          
        </div>
        <div
          className="sidebar__cta mobile-col-2 u-align--right"
        >
          <ul
            className="p-inline-list u-no-margin--top"
          >
            <li
              className="p-inline-list__item"
            >
              <i
                aria-hidden="false"
                className="p-icon--menu "
                onClick={[Function]}
              />
              <i
                aria-hidden="true"
                className="p-icon--close u-hide"
                onClick={[Function]}
              />
            </li>
          </ul>
        </div>
      </div>
    </div>
  </header>
  <div
    className="sidebar__content u-no-margin--top u-hide"
  >
    <ul
      className="p-list"
    >
      <li>
        <a
          className="sidebar__link"
          href="#"
          onClick={[Function]}
        >
          Group
          <i
            className="p-icon--plus"
          />
          <i
            className="p-icon--minus"
          />
        </a>
        <ul
          className="sidebar__second-level"
        >
          <li>
            <a
              className="sidebar__link"
              href="#"
            >
              Link
            </a>
          </li>
        </ul>
      </li>
    </ul>
  </div>
</div>
`;

exports[`<SideNav> renders with a logo-based SideNavBanner correctly 1`] = `
<div
  className="p-navigation--sidebar"
>
  <header
    id="sidenav"
    role="banner"
  >
    <div
      className="p-navigation__banner"
    >
      <div
        className="row"
      >
        <div
          className="mobile-col-2"
        >
          <div
            className="p-navigation__logo"
          >
            <a
              className="p-navigation__link"
              href="#"
            >
              <img
                alt=""
                className="p-navigation__image"
                src="https://assets.ubuntu.com/v1/d96d86b5-vanilla_black-orange_hex.svg"
              />
            </a>
          </div>
          
        </div>
        <div
          className="sidebar__cta mobile-col-2 u-align--right"
        >
          <ul
            className="p-inline-list u-no-margin--top"
          >
            <li
              className="p-inline-list__item"
            >
              <i
                aria-hidden="false"
                className="p-icon--menu "
                onClick={[Function]}
              />
              <i
                aria-hidden="true"
                className="p-icon--close u-hide"
                onClick={[Function]}
              />
            </li>
          </ul>
        </div>
      </div>
    </div>
  </header>
  <div
    className="sidebar__content u-no-margin--top u-hide"
  >
    <ul
      className="p-list"
    />
  </div>
</div>
`;

exports[`<SideNav> renders with a single SideNavLink correctly 1`] = `
<div
  className="p-navigation--sidebar"
>
  <header
    id="sidenav"
    role="banner"
  >
    <div
      className="p-navigation__banner"
    >
      <div
        className="row"
      >
        <div
          className="mobile-col-2"
        >
          <div
            className="p-navigation__logo"
          >
            <a
              className="p-navigation__link"
              href="#"
            >
              Title
            </a>
          </div>
          
        </div>
        <div
          className="sidebar__cta mobile-col-2 u-align--right"
        >
          <ul
            className="p-inline-list u-no-margin--top"
          >
            <li
              className="p-inline-list__item"
            >
              <i
                aria-hidden="false"
                className="p-icon--menu "
                onClick={[Function]}
              />
              <i
                aria-hidden="true"
                className="p-icon--close u-hide"
                onClick={[Function]}
              />
            </li>
          </ul>
        </div>
      </div>
    </div>
  </header>
  <div
    className="sidebar__content u-no-margin--top u-hide"
  >
    <ul
      className="p-list"
    >
      <li>
        <a
          className="sidebar__link"
          href="#"
        >
          Link1
        </a>
      </li>
    </ul>
  </div>
</div>
`;

exports[`<SideNav> renders with a text-based SideNavBanner correctly 1`] = `
<div
  className="p-navigation--sidebar"
>
  <header
    id="sidenav"
    role="banner"
  >
    <div
      className="p-navigation__banner"
    >
      <div
        className="row"
      >
        <div
          className="mobile-col-2"
        >
          <div
            className="p-navigation__logo"
          >
            <a
              className="p-navigation__link"
              href="#"
            >
              Title
            </a>
          </div>
          
        </div>
        <div
          className="sidebar__cta mobile-col-2 u-align--right"
        >
          <ul
            className="p-inline-list u-no-margin--top"
          >
            <li
              className="p-inline-list__item"
            >
              <i
                aria-hidden="false"
                className="p-icon--menu "
                onClick={[Function]}
              />
              <i
                aria-hidden="true"
                className="p-icon--close u-hide"
                onClick={[Function]}
              />
            </li>
          </ul>
        </div>
      </div>
    </div>
  </header>
  <div
    className="sidebar__content u-no-margin--top u-hide"
  >
    <ul
      className="p-list"
    />
  </div>
</div>
`;

exports[`<SideNav> renders with multiple SideNavGroups correctly 1`] = `
<div
  className="p-navigation--sidebar"
>
  <header
    id="sidenav"
    role="banner"
  >
    <div
      className="p-navigation__banner"
    >
      <div
        className="row"
      >
        <div
          className="mobile-col-2"
        >
          <div
            className="p-navigation__logo"
          >
            <a
              className="p-navigation__link"
              href="#"
            >
              Title
            </a>
          </div>
          
        </div>
        <div
          className="sidebar__cta mobile-col-2 u-align--right"
        >
          <ul
            className="p-inline-list u-no-margin--top"
          >
            <li
              className="p-inline-list__item"
            >
              <i
                aria-hidden="false"
                className="p-icon--menu "
                onClick={[Function]}
              />
              <i
                aria-hidden="true"
                className="p-icon--close u-hide"
                onClick={[Function]}
              />
            </li>
          </ul>
        </div>
      </div>
    </div>
  </header>
  <div
    className="sidebar__content u-no-margin--top u-hide"
  >
    <ul
      className="p-list"
    >
      <li>
        <a
          className="sidebar__link"
          href="#"
          onClick={[Function]}
        >
          Group1
          <i
            className="p-icon--plus"
          />
          <i
            className="p-icon--minus"
          />
        </a>
        <ul
          className="sidebar__second-level"
        >
          <li>
            <a
              className="sidebar__link"
              href="#"
            >
              Link1
            </a>
          </li>
        </ul>
      </li>
      <li>
        <a
          className="sidebar__link"
          href="#"
          onClick={[Function]}
        >
          Group2
          <i
            className="p-icon--plus"
          />
          <i
            className="p-icon--minus"
          />
        </a>
        <ul
          className="sidebar__second-level"
        >
          <li>
            <a
              className="sidebar__link"
              href="#"
            >
              Link2
            </a>
          </li>
        </ul>
      </li>
      <li>
        <a
          className="sidebar__link"
          href="#"
          onClick={[Function]}
        >
          Group3
          <i
            className="p-icon--plus"
          />
          <i
            className="p-icon--minus"
          />
        </a>
        <ul
          className="sidebar__second-level"
        >
          <li>
            <a
              className="sidebar__link"
              href="#"
            >
              Link3
            </a>
          </li>
        </ul>
      </li>
    </ul>
  </div>
</div>
`;

exports[`<SideNav> renders with multiple SideNavLinks correctly 1`] = `
<div
  className="p-navigation--sidebar"
>
  <header
    id="sidenav"
    role="banner"
  >
    <div
      className="p-navigation__banner"
    >
      <div
        className="row"
      >
        <div
          className="mobile-col-2"
        >
          <div
            className="p-navigation__logo"
          >
            <a
              className="p-navigation__link"
              href="#"
            >
              Title
            </a>
          </div>
          
        </div>
        <div
          className="sidebar__cta mobile-col-2 u-align--right"
        >
          <ul
            className="p-inline-list u-no-margin--top"
          >
            <li
              className="p-inline-list__item"
            >
              <i
                aria-hidden="false"
                className="p-icon--menu "
                onClick={[Function]}
              />
              <i
                aria-hidden="true"
                className="p-icon--close u-hide"
                onClick={[Function]}
              />
            </li>
          </ul>
        </div>
      </div>
    </div>
  </header>
  <div
    className="sidebar__content u-no-margin--top u-hide"
  >
    <ul
      className="p-list"
    >
      <li>
        <a
          className="sidebar__link"
          href="#"
        >
          Link1
        </a>
      </li>
      <li>
        <a
          className="sidebar__link"
          href="#"
        >
          Link2
        </a>
      </li>
      <li>
        <a
          className="sidebar__link"
          href="#"
        >
          Link3
        </a>
      </li>
    </ul>
  </div>
</div>
`;
